hack/use-openshift-prometheus.sh

```diff
@@ -96,35 +100,38 @@ RESOURCE_LABEL_VALUE="kiali"
 RESOURCE_LABEL_EQUALS="${RESOURCE_LABEL_NAME}=${RESOURCE_LABEL_VALUE}"
 RESOURCE_LABEL_COLON="${RESOURCE_LABEL_NAME}: ${RESOURCE_LABEL_VALUE}"
 
-KIALI_CR_NAMESPACE=""
 KIALI_CR_NAME=""
-for cr in $(${OC} get kiali --all-namespaces -o custom-columns=NS:.metadata.namespace,N:.metadata.name --no-headers | sed 's/  */:/g'); do
-  if [ -n "${KIALI_CR_NAME}" ]; then
-    echo "There is more than one Kiali CR installed in the cluster - will ignore CR [${cr}] and will use [${KIALI_CR_NAMESPACE}:${KIALI_CR_NAME}]"
-  else
+if [ -n "${KIALI_CR_NAMESPACE}" ]
+then
+  for cr in $(${OC} get kiali -n ${KIALI_CR_NAMESPACE} -o custom-columns=N:.metadata.name --no-headers); do
+    KIALI_CR_NAME="${cr}"
+    break
+  done
+  if [ -z "${KIALI_CR_NAME}" ]
+  then
+    echo "Couldn't find any Kiali CR in ${KIALI_CR_NAMESPACE} namespace. Exiting."
+    exit 1
+  fi
+else
+  for cr in $(${OC} get kiali --all-namespaces -o custom-columns=NS:.metadata.namespace,N:.metadata.name --no-headers | sed 's/  */:/g'); do
     KIALI_CR_NAMESPACE="$(echo $cr | cut -d: -f1)"
     KIALI_CR_NAME="$(echo $cr | cut -d: -f2)"
+    break
+  done
+  if [ -z "${KIALI_CR_NAME}" ]; then
+    echo "There are no Kiali CRs installed in the cluster."
+    exit 1
   fi
-done
-
-if [ -z "${KIALI_CR_NAME}" ]; then
-  echo "There are no Kiali CRs installed in the cluster."
-  exit 1
 fi
 
-KIALI_SA_NAMESPACE=""
-KIALI_SA_NAME=""
-for sa in $(${OC} get sa --all-namespaces -l "app.kubernetes.io/name=kiali" -o custom-columns=NS:.metadata.namespace,N:.metadata.name --no-headers | sed 's/  */:/g'); do
-  if [ -n "${KIALI_SA_NAME}" ]; then
-    echo "There is more than one Kiali Server installed in the cluster - ignoring service account [${sa}] and will use [${KIALI_SA_NAMESPACE}:${KIALI_SA_NAME}]"
-  else
-    KIALI_SA_NAMESPACE="$(echo $sa | cut -d: -f1)"
-    KIALI_SA_NAME="$(echo $sa | cut -d: -f2)"
-  fi
-done
+echo "Using following Kiali CR [${KIALI_CR_NAMESPACE}:${KIALI_CR_NAME}]"
 
+# Get a Kiali server namespace, if not defined in .spec.deployment.namespace, use Kiali CR's namespace
+KIALI_SA_NAMESPACE=$(${OC} get kiali ${KIALI_CR_NAME} -n ${KIALI_CR_NAMESPACE} -o jsonpath='{.spec.deployment.namespace}{"\n"}{.metadata.namespace}' | awk 'NF{print $1; exit}')
+# Get a service account requied to generate Kiali Cluster Role Binding for OpenShift Prometheus access
+KIALI_SA_NAME=$(${OC} get sa ${KIALI_CR_NAME}-service-account -n ${KIALI_SA_NAMESPACE} -o custom-columns=N:.metadata.name --no-headers)
```

Comment:
The Kiali CR name has nothing to do with the names of the resources. It is determined from the Kiali CR's `.spec.deployment.instance_name` - if that is not defined, the default is `kiali`.

Reply:
That just returns the string of the expected name... that line doesn't actually attempt to see if it exists. I think that is something you are also trying to do here, so that's something you would need to add back in.